include/cpp_core/serial.h

```diff
@@ -0,0 +1,393 @@
+#pragma once
+#include "version.h"
+
+#include <cstdint>
+
+#if defined(_WIN32) || defined(__CYGWIN__)
+#ifdef cpp_windows_bindings_EXPORTS
+#define MODULE_API __declspec(dllexport)
+#else
+#define MODULE_API __declspec(dllimport)
+#endif
+#else
+#define MODULE_API __attribute__((visibility("default")))
+#endif
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    /**
+     * @brief Copy the compile-time version of the cpp_core library.
+     *
+     * Writes the version baked into the library at build-time to the structure
+     * pointed to by @p out. If @p out is `nullptr` the call is a no-op.
+     *
+     * @param[out] out Pointer to a `cpp_core::Version` structure that receives
+     *                 the version information. May be `nullptr`.
+     */
+    inline MODULE_API void getVersion(cpp_core::Version *out)
+    {
+        if (out != nullptr)
+        {
+            *out = cpp_core::kVersion;
+        }
+    }
+
+    /**
+     * @brief Open and configure a serial port.
+     *
+     * The function attempts to open the device referenced by @p port and applies
+     * the given line settings. The pointer is interpreted as
+     *   • `const char*` on POSIX and
+     *   • `const wchar_t*` on Windows.
+     *
+     * @param port       Null-terminated device identifier (e.g. "COM3",
+     *                   "/dev/ttyUSB0"). Passing `nullptr` results in
+     *                   ::cpp_core::StatusCodes::kNotFoundError.
+     * @param baudrate   Desired baud rate in bit/s (≥ 300).
+     * @param data_bits  Number of data bits (5–8).
+     * @param parity     0 = none, 1 = even, 2 = odd.
+     * @param stop_bits  0 = 1 stop bit, 2 = 2 stop bits.
+     * @return           A positive opaque handle on success or a negative value
+     *                   from ::cpp_core::StatusCodes on failure.
```

Comment:
Could we please get entirely rid of these "indentations" as well, so that the format looks like this:

```suggestion
     * @param port Null-terminated device identifier (e.g. "COM3", "/dev/ttyUSB0").
     *   Passing `nullptr` results in ::cpp_core::StatusCodes::kNotFoundError.
     * @param baudrate Desired baud rate in bit/s (≥ 300).
     * @param data_bits Number of data bits (5–8).
     * @param parity 0 = none, 1 = even, 2 = odd.
     * @param stop_bits 0 = 1 stop bit, 2 = 2 stop bits.
     * @return A positive opaque handle on success or a negative value from
     *   ::cpp_core::StatusCodes on failure.
```